app/controlplane/internal/usercontext/currentuser_middleware.go

```diff
@@ -110,8 +111,44 @@ func WithCurrentUserAndOrgMiddleware(userUseCase biz.UserOrgFinder, logger *log.
 	}
 }
 
-// Find organization and user in DB
-func setCurrentOrgAndUser(ctx context.Context, userUC biz.UserOrgFinder, userID string, logger *log.Helper) (context.Context, error) {
+func WithCurrentOrganizationMiddleware(userUseCase biz.UserOrgFinder, logger *log.Helper) middleware.Middleware {
```

Comment:
I've tried it in the first place but it requires a larger refactor in all middleware. Right now some structs here are so coupled to for example de API token middleware that moving that part requires everything to be properly separated to avoid, as I encountered, import cycles.

Reply:
that's why I mentioned two files not to modules. The can be in the same package?

Reply:
I'll try to reorganize everything.

Reply:
Not much, I'd just put the two middleware in two different files in the same package, no further changes than that.

Reply:
hmm ok I'll do the change but I thought the confusion was because they two were being loaded from the same package and reading it on the grpc middleware list could be confusing.